Sources/App/Core/Mastodon.swift

```diff
@@ -25,7 +25,8 @@ enum Mastodon {
         var accessToken: String
     }
 
-    static func post(client: Client, message: String) async throws {
+    // NG: _testEncodedURL is a callback that exists purely to be able to regression test the encoded value
+    static func post(client: Client, message: String, _testEncodedURL: (String) -> Void = { _ in }) async throws {
```

Comment:
Unfortunately we're leaking this test callback into the production interface but I can't come up with a better way to capture and validate the encoded url before posting. (Brining up a whole fake endpoint to post to feels a bit much.)

We also can't `#ifdef DEBUG` out the parameter itself. We'd have to duplicate the whole function but then we're maintaining two implementations, which is bad.